sensor/common/registry/lazy_tls_image_registry_test.go

```diff
@@ -0,0 +1,283 @@
+	assert.False(t, reg.isInitialized())
+
+	imgName, _, err = utils.GenerateImageNameFromString("example.net/repo/path:latest")
+	require.NoError(t, err)
+	assert.False(t, reg.Match(imgName))
+	assert.False(t, reg.isInitialized())
+}
+
+func TestMetadata(t *testing.T) {
+	cImg, err := utils.GenerateImageFromString("example.com/repo/path:latest")
+	require.NoError(t, err)
+	img := imgTypes.ToImage(cImg)
+
+	t.Run("error when TLS check fails", func(t *testing.T) {
+		source := genImageIntegration("example.com")
+
+		reg, err := createReg(source, nil, alwaysFailCheckTLS)
+		require.NoError(t, err)
+
+		m, err := reg.Metadata(img)
+		require.ErrorContains(t, err, "fake")
+		assert.Nil(t, m)
+		// Confirm that initialization is NOT considered complete due to TLS checks
+		// being temporal.
+		assert.False(t, reg.isInitialized())
+
+		// Subsequent call should skip the TLS check.
+		m, err = reg.Metadata(img)
+		require.ErrorContains(t, err, "skipped")
+		assert.Nil(t, m)
+	})
+
+	t.Run("error when creator fails", func(t *testing.T) {
+		creator := func(integration *storage.ImageIntegration, options ...types.CreatorOption) (types.Registry, error) {
+			return nil, errors.New("fake creator error")
+		}
+		source := genImageIntegration("example.com")
+
+		reg, err := createReg(source, creator, alwaysInsecureCheckTLS)
+		require.NoError(t, err)
+
+		m, err := reg.Metadata(img)
+		require.ErrorContains(t, err, "fake creator error")
+		assert.Nil(t, m)
+
+		// Confirm that initialization IS considered complete.
+		assert.True(t, reg.isInitialized())
+
+		// Repeat to make sure same error is returned.
+		m, err = reg.Metadata(img)
+		require.ErrorContains(t, err, "fake creator error")
+		assert.Nil(t, m)
+	})
+
+	t.Run("successful lazy init", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		mockReg := regMocks.NewMockRegistry(ctrl)
+		fakeMetadata := &storage.ImageMetadata{}
+		creator := func(integration *storage.ImageIntegration, options ...types.CreatorOption) (types.Registry, error) {
+			return mockReg, nil
+		}
+
+		source := genImageIntegration("example.com")
+
+		reg, err := createReg(source, creator, alwaysInsecureCheckTLS)
+		require.NoError(t, err)
+
+		mockReg.EXPECT().Metadata(img).Return(fakeMetadata, nil)
+		m, err := reg.Metadata(img)
+		require.NoError(t, err)
+		assert.True(t, reg.isInitialized())
+		protoassert.Equal(t, fakeMetadata, m)
+
+		mockReg.EXPECT().Metadata(img).Return(fakeMetadata, nil)
+		m, err = reg.Metadata(img)
+		require.NoError(t, err)
+		protoassert.Equal(t, fakeMetadata, m)
+	})
+}
+
+func TestName(t *testing.T) {
+	ii := genImageIntegration("example.com")
+	reg, err := createReg(ii, nil, nil)
+	require.NoError(t, err)
+	assert.Equal(t, "fake-name", reg.Name())
+}
+
+func TestSource(t *testing.T) {
+	ii := genImageIntegration("example.com")
+	reg, err := createReg(ii, nil, nil)
+	require.NoError(t, err)
+
+	got := reg.Source()
+	if got != ii {
+		t.Error(t, "Recieved different datasource than expected.")
+	}
```

Comment:
[nit] Same here. Why not `assert.Equal`?

Reply:
See other comment